src/main/java/TaskManager.java

```diff
@@ -15,43 +17,60 @@ public String markTask(int id, boolean isDone) throws DukeException {
         }
     }
 
+    public String addTask(Task task) {
+        taskList.addTask(task);
+        return Ui.addTaskMsg(task, taskList.getSize());
+    }
+
+    public String notAddDuplicate(Task task) {
+        return Ui.dupTaskMsg(task, taskList.getSize());
+    }
+
     public String addTodo(String[] taskDescription) {
-        Task task = new Todo(taskDescription[0]);
-        return addTask(task);
+        Todo todo = new Todo(taskDescription[0]);
+        if (isDuplicate(todo).isPresent()) {
+            return notAddDuplicate(isDuplicate(todo).get());
+        }
+        return addTask(todo);
     }
 
     public String addDeadline(String[] taskDescription) throws DukeException {
         if (!taskDescription[1].equals("/by")) {
-            throw new DukeException(Ui.inputInWrongFormat());
+            throw new DukeException(Ui.wrongInputFormat());
         }
         if (taskDescription[2].isEmpty()) {
             throw new DukeException(Ui.missingDate());
         }
-        Task task = new Deadline(taskDescription[0], taskDescription[2]);
-        return addTask(task);
+        Deadline deadline = new Deadline(taskDescription[0], taskDescription[2]);
+        if (isDuplicate(deadline).isPresent()) {
+            return notAddDuplicate(isDuplicate(deadline).get());
+        }
+        return addTask(deadline);
     }
 
     public String addEvent(String[] taskDescription) throws DukeException {
         if (!taskDescription[1].equals("/at")) {
-            throw new DukeException(Ui.inputInWrongFormat());
+            throw new DukeException(Ui.wrongInputFormat());
         }
         if (taskDescription[2].isEmpty()) {
             throw new DukeException(Ui.missingDate());
         }
-        Task task = new Event(taskDescription[0], taskDescription[2]);
-        return addTask(task);
-    }
-
-    public String delTask(int id) {
-        int index = id - 1;
-        Task task = taskList.getTask(index);
-        taskList.delTask(task);
-        return Ui.delTaskMsg(task, taskList.getSize());
+        Event event = new Event(taskDescription[0], taskDescription[2]);
+        if (isDuplicate(event).isPresent()) {
+            return notAddDuplicate(isDuplicate(event).get());
+        }
+        return addTask(event);
     }
 
-    public String addTask(Task task) {
-        taskList.addTask(task);
-        return Ui.addTaskMsg(task, taskList.getSize());
+    public String delTask(int id) throws DukeException {
+        try {
+            int index = id - 1;
+            Task task = taskList.getTask(index);
+            taskList.delTask(index);
+            return Ui.delTaskMsg(task, taskList.getSize());
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeException(Ui.taskIdOutOfBound(id));
```

Comment:
Maybe you could let IndexOutOfBound handle the exception instead of throwing it to Duke to handle.